### DISCUSSION TITLE
Behavior of `CCIO.File.read_dir ~recurse:true`

### DISCUSSION THREAD

**Discussion by FardaleM** [c-cube/ocaml-containers#391] [General]
The function `CCIO.File.read_dir` list directory if `recurse` is `false` but does not if `recurse` is `true`. That's quite a surprising behavior. Maybe we should just split into another function to have the second behavior.

**Comment by c-cube** [comment]
I would consider that to be a bug, and thus amenable to a breaking change :-).

**Comment by FardaleM** [comment]
I see, what should be the behavior of `CCIO.File.read_dir recurse:true`? Should it output everything including the directories?

**Comment by FardaleM** [comment]
Just so that I don't forget, this behavior was mention here https://github.com/c-cube/ocaml-containers/blob/a127e139aeb48b12b16b8a928cefcf54f8bced5c/src/core/CCIO.mli#L186 but not in the doc of the function.

**Reply by c-cube** [reply]
that's a good point, I forgot that the functionality exists already. Maybe we could just update the docs then.
